tests/jballerina-unit-test/src/test/java/org/ballerinalang/test/parser/InvalidSyntaxParserTest.java

```diff
@@ -112,4 +112,32 @@ public void testTokenRecognition() {
         BAssertUtil.validateError(result, 0, "token recognition error at: '\\'", 3, 5);
         BAssertUtil.validateError(result, 0, "token recognition error at: '*'", 4, 15);
     }
+
+    @Test
+    public void testListenerDeclarationWithoutType() {
+        CompileResult result = BCompileUtil.compile("test-src/parser/listener-declaration-without-type-negative.bal");
+        BAssertUtil.validateError(result, 0, "incompatible types: expected 'Listener', found 'other'", 17, 14);
```

Comment:
Shouldn't the error be

```suggestion
        BAssertUtil.validateError(result, 0, "incompatible types: expected 'lang.object:Listener', found 'other'", 17, 14);
```

instead?